pkg/kv/kvserver/kvserverpb/state.proto

```diff
@@ -90,9 +90,8 @@ message ReplicaState {
   // (#72222). Once that's done, we should be able to stop sending the replica
   // state as part of the snapshot in the subsequent release.
   //
-  // TODO(irfansharif): Remove this field in the release after the one where we
-  // stop consulting the protobuf copy of the replica state when applying a
-  // snapshot.
+  // This field can be removed together with
```

Comment:
The comment here is interesting, btw. We *thought* we had figured out when this was safe to remove, but it wasn't actually true yet. No amount of version gating would've fixed that.